src/screens/Feed.js

```diff
@@ -1,124 +1,122 @@
-/* eslint-disable react-native/no-inline-styles */
 import React from 'react';
-import {View, StyleSheet, FlatList} from 'react-native';
+import {
+  View,
+  StyleSheet,
+  FlatList,
+  TouchableWithoutFeedback,
+  Text,
+  SafeAreaView,
+  Alert,
+} from 'react-native';
+
 import SettingsButton from './app/Components/SettingsButton/SettingsButton';
 import ProfileAvatar from './app/Components/ProfileAvatar/ProfileAvatar';
 import Post from './app/Components/Post/Post';
+import Swiper from 'react-native-swiper';
+import {Divider} from 'react-native-paper';
+import firestore from '@react-native-firebase/firestore';
+import * as ScreenshotDetector from 'react-native-screenshot-detect';
+
 export default class FeedScreen extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      swiperIndex: 0,
+      posts: [],
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchPosts();
+
+    this.eventEmitter = ScreenshotDetector.subscribe(() => {
+      Alert.alert(
+        'The post with caption',
+        posts[this.state.swiperIndex].caption,
+        'was screenshotted',
+      );
+    });
+  }
+
+  componentWillUnmount() {
+    ScreenshotDetector.unsubscribe(this.eventEmitter);
   }
 
+  fetchPosts = () => {
+    const subscriber = firestore()
+      .collection('posts')
+      .onSnapshot(querySnapshot => {
+        const posts = [];
+
+        querySnapshot.forEach(documentSnapshot => {
+          posts.push({
+            ...documentSnapshot.data(),
+            key: documentSnapshot.id,
+          });
```

Comment:
Is this pulling all relevant posts? Thinking about scalability, if a user has 500 posts on their feed, we would want to render only 10? 

Can we somehow create a callback from FlatList, which pulls in posts in a batch, after each one has been previewed?